osu.Game/Utils/StatelessRNG.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Utils
+{
+    /// Provides a fast stateless function that can be used in randomly-looking visual elements.
+    public static class StatelessRNG
+    {
+        private static ulong mix(ulong x)
+        {
+            unchecked
+            {
+                x ^= x >> 33;
+                x *= 0xff51afd7ed558ccd;
+                x ^= x >> 33;
+                x *= 0xc4ceb9fe1a85ec53;
+                x ^= x >> 33;
+                return x;
+            }
+        }
+
+        /// Compute an integer from given seed and series number.
+        /// <param name="seed">
+        /// The seed value of this random number generator.
+        /// </param>
+        /// <param name="series">
+        /// The series number.
+        /// Different values are computed for the same seed in different series.
+        /// </param>
+        public static ulong Get(int seed, int series = 0) =>
+            unchecked(mix(((ulong)(uint)series << 32) | ((uint)seed ^ 0x12345678)));
```

Comment:
Is this double `(ulong)(uint)` cast necessary?

Reply:
It is necessary, because `(ulong)-1` is `0xffffffffffffffff`, while `(ulong)(uint)-1` is `0x00000000ffffffff`.